### PR TITLE
itm: replace itm-tools

### PR DESCRIPTION
**Pull request by mrshmllow**
## Description of changes

[itm-tools](https://github.com/japaric/itm-tools) (@sbourdeauducq @HarryMakes) has been replaced by [itm](https://github.com/rtic-scope/itm) since early 2022.

itm-tools:
> This repository is no longer maintained!
> The library on which these tools are based has been deprecated in favor of [rtic-scope/itm].
> You may want to check the tools developed by the [rtic-scope] team.

(See https://github.com/japaric/itm-tools/commit/583992a7fd3d1e293f8dca0944c0fe83dc378edd)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/it/itm/package.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  meta = with lib; {
+    mainProgram = "itm-decode";
+    description = "ARMv7-M ITM packet protocol decoder library crate and CLI tool.";
+    homepage = "https://github.com/rtic-scope/itm";
+    license = with licenses; [asl20 mit];
+    maintainers = with maintainers; [
+      marshmallow
+    ];
+  };
```
Comment (itslychee) [inline pkgs/by-name/it/itm/package.nix:40]: A nitpick but personally I think it'd look better to remove the `with lib;` and use it like `with lib.licenses`, `with lib.maintainers`, etc though it really is up to personal preference.

**Comment by itslychee** [comment]
Result of `nixpkgs-review pr 285957` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>itm</li>
  </ul>
</details>

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 285957` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>itm</li>
  </ul>
</details>

File: pkgs/by-name/it/itm/package.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  nativeBuildInputs = [pkg-config];
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  meta = with lib; {
+    mainProgram = "itm-decode";
+    description = "ARMv7-M ITM packet protocol decoder library crate and CLI tool.";
```
Comment (applePrincess) [inline pkgs/by-name/it/itm/package.nix:34]: Description shold not end with a period

**Review by itslychee — APPROVED** [review]

**Comment by mrshmllow** [comment]
Closing what little contribution I have because of https://discourse.nixos.org/t/should-jonringer-get-his-commit-bit-back/47267

**Comment by applePrincess** [comment]
I do not want to make a statement about the thread you mention but this PR is not a small contribution and adding yourself to `maintainer-list.nix` is not enough to get committer right.
If that is what you are worrying about then I think it is fine. Or just not interested in anymore?

**Comment by mrshmllow** [comment]
I had no expectation to become a committer, no, lol. I dont want to contribute to this repo anymore.

**Pull request closed without merging**